src/Facets/EverclearFacet.sol

```diff
@@ -0,0 +1,235 @@
+
+    /// Constructor ///
+
+    /// @notice Constructor for the contract.
+    /// @param _feeAdapter Fee adapter address.
+    constructor(address _feeAdapter) {
+        // Validate fee adapter address to prevent misconfiguration that would break bridge functionality
+        if (address(_feeAdapter) == address(0)) {
+            revert InvalidConfig();
+        }
+        FEE_ADAPTER = IEverclearFeeAdapter(_feeAdapter);
+    }
+
+    /// External Methods ///
+
+    /// @notice Bridges tokens via Everclear
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _everclearData Data specific to Everclear
+    function startBridgeTokensViaEverclear(
+        ILiFi.BridgeData memory _bridgeData,
+        EverclearData calldata _everclearData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        noNativeAsset(_bridgeData)
+    {
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+        _startBridge(_bridgeData, _everclearData);
+    }
+
+    /// @notice Performs a swap before bridging via Everclear
+    /// @param _bridgeData The core information needed for bridging
+    /// @param _swapData An array of swap related data for performing swaps before bridging
+    /// @param _everclearData Data specific to Everclear
+    function swapAndStartBridgeTokensViaEverclear(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        EverclearData calldata _everclearData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+        noNativeAsset(_bridgeData)
+    {
+        uint256 actualAmountAfterSwap = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+
+        // If we got more than expected from the swap (positive slippage),
+        // send the extra back to the user. We cannot adjust amountOutMin
+        // because Everclear's signature validation includes the original value.
+        if (actualAmountAfterSwap > _bridgeData.minAmount) {
+            uint256 positiveSlippage = actualAmountAfterSwap -
+                _bridgeData.minAmount;
+            LibAsset.transferERC20(
+                _bridgeData.sendingAssetId,
+                payable(msg.sender),
+                positiveSlippage
+            );
+            // Bridge only the original expected amount
+            actualAmountAfterSwap = _bridgeData.minAmount;
+        }
+
+        _bridgeData.minAmount = actualAmountAfterSwap;
```

Comment:
actualAmountAfterSwap cannot be lower than _bridgeData.minAmount since _depositAndSwap function will ensure the minAmount.

So we have two cases: 
Case 1: `actualAmountAfterSwap == _bridgeData.minAmount` (no slippage)
Case 2: `actualAmountAfterSwap > _bridgeData.minAmount` (positive slippage)

in both cases `_bridgeData.minAmount` should remain the same, I dont think it makes sense to update it. 
Also line 136 can be removed imho.